## Basic Understanding - 

Screen Time Analysis is the task of analyzing and creating a report on which applications and websites are used by the user for how much time. It lets you know how much time you spend on what kind of applications and websites using your device and gives a visual report of the same.

## Dataset - 


The dataset contains data about:

1) Date
2) Usage of Applications 
3) Number of Notifications from Applications 
4) Number of times apps opened


## Importing Dependencies - 

In [1]:
import pandas as pd
import numpy as np
import plotly.express as px
import seaborn as sns
sns.set()

In [2]:
df = pd.read_csv('./data/Screentime-App-Details.csv')

In [3]:
df.head()

Date  Usage  Notifications  Times opened        App
0  08/26/2022     38             70            49  Instagram
1  08/27/2022     39             43            48  Instagram
2  08/28/2022     64            231            55  Instagram
3  08/29/2022     14             35            23  Instagram
4  08/30/2022      3             19             5  Instagram

## Exploring Dataset - 

In [4]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 54 entries, 0 to 53
Data columns (total 5 columns):
 #   Column         Non-Null Count  Dtype 
---  ------         --------------  ----- 
 0   Date           54 non-null     object
 1   Usage          54 non-null     int64 
 2   Notifications  54 non-null     int64 
 3   Times opened   54 non-null     int64 
 4   App            54 non-null     object
dtypes: int64(3), object(2)
memory usage: 2.2+ KB


#### Let’s have a look at the descriptive statistics of the data.


In [5]:
df.describe()

Usage  Notifications  Times opened
count   54.000000      54.000000     54.000000
mean    65.037037     117.703704     61.481481
std     58.317272      97.017530     43.836635
min      1.000000       8.000000      2.000000
25%     17.500000      25.750000     23.500000
50%     58.500000      99.000000     62.500000
75%     90.500000     188.250000     90.000000
max    244.000000     405.000000    192.000000

#### Lets check for any null values

In [6]:
df.isnull().sum()

Date             0
Usage            0
Notifications    0
Times opened     0
App              0
dtype: int64

There are no null values in our dataset, so we can proceed with visualising the data. 

## Visualization - 

#### Finding out how many applications we are working with -

In [7]:
df["App"].unique()

array(['Instagram', 'Whatsapp'], dtype=object)

We have 2 applications - Instagram and Whatsapp.

#### Lets find the start and end dates in our dataset -

In [8]:
df["Date"].head()

0    08/26/2022
1    08/27/2022
2    08/28/2022
3    08/29/2022
4    08/30/2022
Name: Date, dtype: object

In [9]:
df["Date"].tail()

49    09/17/2022
50    09/18/2022
51    09/19/2022
52    09/20/2022
53    09/21/2022
Name: Date, dtype: object

#### Finding the total usage from our dataset - 

In [10]:
df["Usage"].sum()

3512

 We can see that the total usage between the 2 applications from 08/26/2022 to 09/21/2022 is 3512 hours. We can plot the same using a bar graph in the following manner - 

In [11]:
figure  = px.bar(data_frame = df,
                   x = "Date",
                   y = "Usage",
                   color = "App",
                   title = "Total time spent on different applications")

figure.show()

Now lets look at the number of notifications we recieve from our apps on a daily basis:

In [12]:
figure = px.bar(data_frame = df,
                y= "Notifications",
                x = "Date",
                color= "App",
                title = "Notifications per application per day")
figure.show()

Now let’s have a look at the number of times the apps opened:

In [13]:
figure = px.bar(data_frame = df,
                x = "Date",
                y= "Times opened",
                color= "App",
                title = "No of time the application was opened per day")
figure.show()

We generally open our applications, whenever we get a notification from it. Lets see the relationship between the number of times we open our application when we get a notification.

In [14]:
figure = px.scatter(data_frame= df,
                    x = "Notifications",
                    y = "Times opened",
                    color = "Times opened",
                    size = "Notifications",
                    trendline = "ols",
                    title = "Relationship Between Number of Notifications and Times Opened. ")
figure.show()

#### Inference - 

There’s a linear relationship between the number of notifications and the number of times opened. It means that more notifications result in more use of the target application.

### Finding relation usage and number of notification

We generally use our smartphones when we get notified by any app. So let’s have a look at the relationship between the number of notifications and the amount of usage: 

In [15]:
figure = px.scatter(data_frame = df, 
                    x="Notifications",
                    y="Usage", 
                    size="Notifications", 
                    color = "Usage",
                    trendline="ols", 
                    title = "Relationship Between Number of Notifications and Usage")
figure.show()

### Inference - 

There’s a linear relationship between the number of notifications and the amount of usage. It means that more notifications result in more use of smartphones.